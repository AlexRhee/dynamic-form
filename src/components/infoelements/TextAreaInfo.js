import React from 'react';

//Element for the text are info (post submission)
const TextAreaInput = (props) => {
    return (
        <div data-testid="textareainfo-1" className="input-wrapper">
            <label>{props.field.name}:</label>
            <div>
                <textarea disabled type="textarea" className="form-control"  placeholder={props.field.placeholder} value={props.field.value}/>
            </div>
        </div>
    )
}

export default TextAreaInput;