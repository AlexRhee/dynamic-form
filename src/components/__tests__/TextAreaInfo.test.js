import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TextAreaInfo from '../infoelements/TextAreaInfo';

afterEach(() => {
    cleanup();
})

test('should render TextAreaInfo component with prefilled input information, and be disabled', () => {

    const field =         {
        "id": "reason",
        "name": "Reason",
        "placeholder": "Describe why you are a good fit for the job you are applying for.",
        "type": "textarea",
        "ind": 9,
        "value": "Tech stacks align, culture seems like a good fit."
    }

    render(<TextAreaInfo field={field} />);
    const textareainfoElement = screen.getByTestId('textareainfo-1');
    expect(textareainfoElement).toBeInTheDocument();
    expect(textareainfoElement).toHaveTextContent('Tech stacks align');
    expect(textareainfoElement).toHaveTextContent('Reason');
    expect(textareainfoElement).toBeDisabled;
});

test('matches snapshot', () => {

    const field =         {
        "id": "reason",
        "name": "Reason",
        "placeholder": "Describe why you are a good fit for the job you are applying for.",
        "type": "textarea",
        "ind": 9,
        "value": "Tech stacks align, culture seems like a good fit."
    }

    const tree = renderer.create(<TextAreaInfo field={field} />).toJSON();
    expect(tree).toMatchSnapshot();
})