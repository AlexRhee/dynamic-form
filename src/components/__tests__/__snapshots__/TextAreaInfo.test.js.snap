// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="input-wrapper"
  data-testid="textareainfo-1"
>
  <label>
    Reason
    :
  </label>
  <div>
    <textarea
      className="form-control"
      disabled={true}
      placeholder="Describe why you are a good fit for the job you are applying for."
      type="textarea"
      value="Tech stacks align, culture seems like a good fit."
    />
  </div>
</div>
`;
